From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Subject: [PATCH v2] combase: Add RoSetErrorReportingFlags stub
Message-Id: <PU4P216MB125532F15B5CD736D117029B933E9@PU4P216MB1255.KORP216.PROD.OUTLOOK.COM>
Date: Fri, 25 Feb 2022 19:19:46 +1100

Rebase against git, as the api-ms-win-core-winrt-error-l1-1-* no longer exist.

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/combase/combase.spec |  2 +-
 dlls/combase/roapi.c      | 10 ++++++++++
 include/roerrorapi.h      | 10 ++++++++++
 3 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/dlls/combase/combase.spec b/dlls/combase/combase.spec
index 11111111111..11111111111 100644
--- a/dlls/combase/combase.spec
+++ b/dlls/combase/combase.spec
@@ -312,7 +312,7 @@
 @ stub RoReportUnhandledError
 @ stub RoResolveRestrictedErrorInfoReference
 @ stub RoRevokeActivationFactories
-@ stub RoSetErrorReportingFlags
+@ stdcall RoSetErrorReportingFlags(long)
 @ stub RoTransformError
 @ stub RoTransformErrorW
 @ stdcall RoUninitialize()
diff --git a/dlls/combase/roapi.c b/dlls/combase/roapi.c
index 11111111111..11111111111 100644
--- a/dlls/combase/roapi.c
+++ b/dlls/combase/roapi.c
@@ -316,3 +316,13 @@ HRESULT WINAPI DllGetActivationFactory(HSTRING classid, IActivationFactory **fac
 
     return REGDB_E_CLASSNOTREG;
 }
+
+/***********************************************************************
+ *      RoSetErrorReportingFlags (combase.@)
+ */
+HRESULT WINAPI RoSetErrorReportingFlags(UINT32 flags)
+{
+    FIXME("%d\n", flags);
+    return S_OK;
+}
+
diff --git a/include/roerrorapi.h b/include/roerrorapi.h
index 11111111111..11111111111 100644
--- a/include/roerrorapi.h
+++ b/include/roerrorapi.h
@@ -23,8 +23,18 @@
 #include <restrictederrorinfo.h>
 #include <rpc.h>
 
+typedef enum
+{
+    None,
+    SuppressExceptions,
+    ForceExceptions,
+    UseSetErrorInfo,
+    SuppressSetErrorInfo
+} RoErrorReportingFlags;
+
 HRESULT WINAPI GetRestrictedErrorInfo(IRestrictedErrorInfo **info);
 BOOL    WINAPI RoOriginateError(HRESULT error, HSTRING message);
 BOOL    WINAPI RoOriginateLanguageException(HRESULT error, HSTRING message, IUnknown *language_exception);
+HRESULT WINAPI RoSetErrorReportingFlags(UINT32 flags);
 
 #endif /* _ROERROR_H */

-- 
2.35.1

